react/features/app/actions.js

```diff
@@ -245,3 +250,20 @@ function _loadConfig({ contextRoot, host, protocol, room }) {
             throw error;
         });
 }
+
+/**
+ * Merges the downloaded config with the current profile values. The profile
+ * values are named the same way as the config values in the config.js so
+ * a clean merge is possible.
+ *
+ * @param {Object|undefined} config - The downloaded config.
+ * @param {Object} profile - The persisted profile.
+ * @returns {Object}
+ */
+function _mergeConfigWithProfile(config, profile) {
+    if (!config) {
+        return;
+    }
+
+    return Object.assign(config, profile);
```

Comment:
Does this modify the redux config in place? I'm not sure how I feel about logic that's changing objects in place, especially when state in redux should be replaced instead. I think in the end it doesn't matter much in this example as the new config will immediately be set as the config in redux, but in general I think staying away from object mutation would be more in line with redux.

Reply:
Good point about object mutation, I fix it. But not sure what you mean about changing the redux config. This part has nothing to do with redux, it just merges the config.js properties that we download with the stored profile, with profile being the higher priority.

Reply:
From what I remember, you were taking the config object that was in redux, the actual object inside the redux store and not a copy, and were mutating that directly with Object.assign instead of going through a reducer that replaced the object.

Reply:
We talked about this offline and I had misread what was being modified in place. What was being modified is the global config, which has a precedence already for being modified in place.